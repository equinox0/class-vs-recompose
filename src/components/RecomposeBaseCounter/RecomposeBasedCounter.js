import React from "react";
import { compose, withHandlers, withState } from "recompose";

const RecomposeBasedCounter = ({ count, handleDec, handleInc }) => (
    <div>
        <div>
            <strong>Counter: </strong>
            {count}
        </div>
        <div>
            <button type="button" onClick={handleInc}>
                +
            </button>
            {` `}
            <button type="button" onClick={handleDec}>
                -
            </button>
        </div>
    </div>
);

const enhance = compose(
    withState("count", "setCount", 0),
    withHandlers({
        handleDec: ({ count, setCount }) => () => setCount(count - 1),
        handleInc: ({ count, setCount }) => () => setCount(count + 1)
    })
);

export { RecomposeBasedCounter, enhance };
export default enhance(RecomposeBasedCounter);
