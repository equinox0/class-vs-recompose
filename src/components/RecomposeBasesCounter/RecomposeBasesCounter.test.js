import { shallow, mount } from "enzyme";
import React from "react";
import { RecomposeBasedCounter, enhance } from "./RecomposeBasedCounter";

describe("<RecomposeBasedCounter/>", () => {
  describe("component", () => {
    const setup = () => {
      const props = {
        count: 0,
        handleDec: jest.fn(),
        handleInc: jest.fn()
      };
      const wrapper = shallow(<RecomposeBasedCounter {...props} />);
      return { wrapper };
    };

    it("should display counter with two buttons", () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("enhance", () => {
    const setup = () => {
      const MockComponent = () => null;
      const EnhancedComponent = enhance(MockComponent);
      const wrapper = mount(<EnhancedComponent />);
      const outputProps = wrapper.find(MockComponent).props();
      return { MockComponent, outputProps, wrapper };
    };

    it("should increment", () => {
      const { outputProps } = setup();
      outputProps.handleInc();
    });

    it("should decrement", () => {
      const { outputProps } = setup();
      outputProps.handleInc();
    });

    it("should have count", () => {
      const { outputProps } = setup();
      expect(outputProps.count).toEqual(0);
    });
  });
});
