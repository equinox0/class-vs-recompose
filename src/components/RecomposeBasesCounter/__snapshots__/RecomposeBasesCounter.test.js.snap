// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecomposeBasedCounter/> component should display counter with two buttons 1`] = `
<div>
  <div>
    <strong>
      Counter: 
    </strong>
    0
  </div>
  <div>
    <button
      onClick={[MockFunction]}
      type="button"
    >
      +
    </button>
     
    <button
      onClick={[MockFunction]}
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
